specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,981 @@
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of the web application firewall policy.",
+          "$ref": "#/definitions/WebApplicationFirewallPolicyProperties"
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        },
+        "sku": {
+          "description": "The pricing tier of web application firewall policy. Defaults to Classic_AzureFrontDoor if not specified.",
+          "$ref": "#/definitions/Sku"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyProperties": {
+      "description": "Defines web application firewall policy properties.",
+      "properties": {
+        "policySettings": {
+          "description": "Describes settings for the policy.",
+          "$ref": "#/definitions/PolicySettings"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy.",
+          "$ref": "#/definitions/CustomRuleList"
+        },
+        "managedRules": {
+          "description": "Describes managed rules inside the policy.",
+          "$ref": "#/definitions/ManagedRuleSetList"
+        },
+        "frontendEndpointLinks": {
+          "description": "Describes Frontend Endpoints associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FrontendEndpointLink"
+          }
+        },
+        "routingRuleLinks": {
+          "description": "Describes Routing Rules associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/RoutingRuleLink"
+          }
+        },
+        "securityPolicyLinks": {
+          "description": "Describes Security Policy associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/SecurityPolicyLink"
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state of the policy."
+        },
+        "resourceState": {
+          "title": "Resource status of the policy.",
+          "readOnly": true,
+          "enum": [
+            "Creating",
+            "Enabling",
+            "Enabled",
+            "Disabling",
+            "Disabled",
+            "Deleting"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PolicyResourceState",
+            "modelAsString": true
+          }
+        },
+        "extendedProperties": {
```

Comment:
>extendedProperties [](http://example.com/codeflow?start=9&length=18)

Generic key value pairs are discouraged. Can you define a proper schema for this?

Reply:
As mentioned in previous comment, we would like to keep it this way in this case. We will stop doing this for any further projects.